### PR TITLE
Feat: Go discord command bot

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

Adds discord bot template in Go language

## Test Plan

- [x] Manual QA 

CleanShot 2024-07-04 at 15 36 09@2x

## Related PRs and Issues

x

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: go/discord-command-bot/.prettierrc.json
Comment (TorstenDittmann) [inline go/discord-command-bot/.prettierrc.json:1]: prettier for golang? 👀

File: go/discord-command-bot/cli/setup.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+		"DISCORD_TOKEN",
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	registerApi := "https://discord.com/api/v9/applications/" + os.Getenv("DISCORD_APPLICATION_ID") + "/commands"
+
+	bodyJson := map[string]string{"name": "hello", "description": "Hello World Command"}
+	bodyString, err := json.Marshal(bodyJson)
```
Comment (TorstenDittmann) [inline go/discord-command-bot/cli/setup.go:26]: we might wanna create a struct for this with field tags 👍🏻

File: go/discord-command-bot/main.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		}, 400, nil)
+	}
+
+	ApplicationCommandType := 2
+	if discordBody.Type == ApplicationCommandType && discordBody.Data.Name == "hello" {
+		Context.Log("Matched hello command - returning message")
+
+		channelMessageWithSource := 4
+		return Context.Res.Json(
+			any{
```
Comment (TorstenDittmann) [inline go/discord-command-bot/main.go:51]: maybe have a struct for this prepared too?

**Review by TorstenDittmann — APPROVED** [review]

File: go/discord-command-bot/cli/setup.go (hunk 1 above)
Comment (Meldiron) [reply go/discord-command-bot/cli/setup.go:26]: Its overkill here I would say

File: go/discord-command-bot/main.go (hunk 2 above)
Comment (Meldiron) [reply go/discord-command-bot/main.go:51]: Its overkill here I think. Discord response can have MANY different params, and I don't want this to be library that defined it.

File: go/discord-command-bot/main.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+				"data": any{
+					"content": "Hello, World!",
+				},
+			},
+			200,
+			nil,
+		)
+	}
+
+	Context.Log("Didn't match command - returning PONG")
```
Comment (Meldiron) [inline go/discord-command-bot/main.go:62]:
```suggestion
	Context.Log("Did not match command - returning PONG")
```

**Pull request merged**